libraries/c_sdk_v4/integration-test/core_mqtt_system_test.c

```diff
@@ -0,0 +1,1635 @@
+    establishMqttSession( &context, &networkContext, false, &persistentSession );
+
+    /* Verify that the session was resumed. */
+    TEST_ASSERT_TRUE( persistentSession );
+}
+
+/* ============================   UNITY FIXTURES ============================ */
+
+/* Called before each test method. */
+void testSetUp()
+{
+    /* Reset file-scoped global variables. */
+    receivedSubAck = false;
+    receivedUnsubAck = false;
+    receivedPubAck = false;
+    receivedPubRec = false;
+    receivedPubRel = false;
+    receivedPubComp = false;
+    receivedRetainedMessage = false;
+    persistentSession = false;
+    useLWTClientIdentifier = false;
+    packetTypeForDisconnection = MQTT_PACKET_TYPE_INVALID;
+    memset( &incomingInfo, 0u, sizeof( MQTTPublishInfo_t ) );
+
+    /* Initializer server information. */
+    serverInfo.pHostName = BROKER_ENDPOINT;
+    serverInfo.hostNameLength = strlen( BROKER_ENDPOINT );
+    serverInfo.port = BROKER_PORT;
+
+    /* Initialize SocketsConfig. */
+    socketsConfig.enableTls = true;
+    socketsConfig.pAlpnProtos = NULL;
+    socketsConfig.maxFragmentLength = 0;
+    socketsConfig.disableSni = true;
+    socketsConfig.pRootCa = SERVER_ROOT_CA_CERT;
+    socketsConfig.rootCaSize = strlen( SERVER_ROOT_CA_CERT ) + 1U;
+    socketsConfig.sendTimeoutMs = TRANSPORT_SEND_RECV_TIMEOUT_MS;
+    socketsConfig.recvTimeoutMs = TRANSPORT_SEND_RECV_TIMEOUT_MS;
+
+    /* Establish a TCP connection with the server endpoint, then
+     * establish TLS session on top of TCP connection. */
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_SUCCESS, SecureSocketsTransport_Connect( &networkContext,
+                                                                                        &serverInfo,
+                                                                                        &socketsConfig ) );
+    TEST_ASSERT_NOT_NULL( networkContext.pContext );
+
+    /* Establish MQTT session on top of the TCP+TLS connection. */
+    establishMqttSession( &context, &networkContext, true, &persistentSession );
+}
+
+/* Called after each test method. */
+void testTearDown()
+{
+    /* Free memory, if allocated during test case execution. */
+    if( incomingInfo.pTopicName != NULL )
+    {
+        vPortFree( ( void * ) incomingInfo.pTopicName );
+    }
+
+    if( incomingInfo.pPayload != NULL )
+    {
+        vPortFree( ( void * ) incomingInfo.pPayload );
+    }
+
+    /* Terminate MQTT connection. */
+    TEST_ASSERT_EQUAL( MQTTSuccess, MQTT_Disconnect( &context ) );
+
+    /* Terminate TLS session and TCP connection. */
+    ( void ) SecureSocketsTransport_Disconnect( &networkContext );
+}
+
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test group for coreMQTT system tests of MQTT 3.1.1
+ * features that are supported by AWS IoT.
+ */
+TEST_GROUP( coreMQTT_Integration_AWS_IoT_Compatible );
+
+TEST_SETUP( coreMQTT_Integration_AWS_IoT_Compatible )
+{
+    testSetUp();
```

Comment:
How is a mutually authenticated connection created with AWS IoT broker?

Reply:
The device client credentials are provided in the `aws_clientcredential_key.h` file which is picked up by the `iot_tls.c` file (that is called by Secure Sockets as we are using the Secure Sockets transport interface implementation). 
This is the existing test infrastructure of this repository for providing client credentials for all tests that connect to AWS IoT